File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin() , players.end());
        sort(trainers.begin(),trainers.end());
        int tot=0,pp=0,tp=0;
        while(pp<players.size() && tp<trainers.size()){
            if (players[pp] <= trainers[tp]) { tot++;pp++;tp++;}
            else tp++;
        }
        return tot;
    }
};

File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players = sorted(players, reverse = True)
        trainers = sorted(trainers, reverse = True)
        i,j= 0,0
        ans = 0
        # print(players , trainers)
        while i<len(players) and j<len(trainers):
            if players[i] <= trainers[j]:
                ans += 1
                j+=1
            i+=1
        return ans